datadog_lambda/tracing.py

```diff
@@ -1309,8 +1321,15 @@ def create_inferred_span_from_eventbridge_event(event, context):
         synchronicity="async",
         tag_source="self",
     )
-    dt_format = "%Y-%m-%dT%H:%M:%SZ"
-    timestamp = event.get("time")
+
+    # Use more granular timestamp from upstream Step Function if possible
+    if is_step_function_event(event.get("detail")):
+        timestamp = event.get("detail").get("_datadog").get("State").get("EnteredTime")
```

Comment:
Is it possible that these keys are not in the event? If so, this will raise an exception which I'm not sure is being caught?

Reply:
It could be possible if a customer messes up the instrumentation

I think it's worth handling tho so lemme add some logic to fallback to `event.get("time")` if this fails